src/Integration/NewConnectedMode/ConfigurationSerializer.cs

```diff
@@ -49,15 +51,37 @@ internal class ConfigurationSerializer : FileBindingSerializer
             ISourceControlledFileSystem sccFileSystem,
             ICredentialStore store,
             ILogger logger,
-            IFile fileWrapper)
+            IFile fileWrapper,
+            IDirectory directoryWrapper)
             :base(sccFileSystem, store, logger, fileWrapper)
         {
             if (solution == null)
             {
                 throw new ArgumentNullException(nameof(solution));
             }
+            Debug.Assert(directoryWrapper != null);
```

Comment:
Why not throw?

Reply:
The internal constructor should only be called from tests. The public constructors supplies a valid value, so it should never be null.

Reply:
Ok